### ISSUE TITLE
Using WikitablesQuestion Semantic Parser Demo programatically 

### ISSUE THREAD

**Issue by shubhamagarwal92** [allenai/allennlp#2771]
Thanks for the nice work! 

I would like to use the WikitablesQuestion Semantic Parser out of the box (as possible in the [demo](https://demo.allennlp.org/wikitables-parser)) from command line. I will be using my own set of tables and questions and want to plug this model to retrieve the result as part of a pipeline model that I am working on. 

1. How to replicate the demo locally through python script? Could you provide me the SOTA model that I could use?
2. What should be the format of the data file containing a table? Could this be a tab/comma separated CSV with header?

**Comment by shubhamagarwal92** [comment]
Following #1816 and #2598, I used [wikitables-model-2018.09.14.tar.gz](https://s3-us-west-2.amazonaws.com/allennlp/models/wikitables-model-2018.09.14.tar.gz) as the pre-trained model to replicate the demo locally.

Data input format that I used is a JSON with `predict_json` function of the predictor. (However, the `predict` function is not working for wikitables semantic parser as described in the tutorials.)

Also for the tables I am using my comma separated CSV file.

@pdasigi Is this the best pre-trained model as of now? The input format of CSV should be comma separated, right? Is this the correct way?

Full code here:
```
import argparse
from allennlp.models.archival import load_archive
from allennlp.predictors import Predictor
# from allennlp.predictors.predictor import Predictor

class DemoModel:
    """
    A demo model is determined by an archive file (representing the trained model), a
    choice of predictor, and a limit on the maximum request length it can handle.

    Expects JSON that looks like {"question": "...", "table": "..."}
    """
    def __init__(self, archive_file: str, predictor_name: str = "wikitables-parser") -> None:
        self.archive_file = archive_file
        self.predictor_name = predictor_name

    def predictor(self) -> Predictor:
        archive = load_archive(self.archive_file)
        return Predictor.from_archive(archive, self.predictor_name)

def main():
    parser = argparse.ArgumentParser(description='Wikitables Semantic Parse')
    parser.add_argument('-model_path', type=str,
                        default="wikitables-model-2018.09.14.tar.gz",
                        help="path to model")
    parser.add_argument('-table_path', type=str, default="",
                        help="path to tables")
    args = parser.parse_args()

    with open(args.table_path, 'r') as file:
        table = file.read()
    print("Table contents: \n", table)
    model = DemoModel(args.model_path)
    predictor = model.predictor()

    input_data = {"table":table,
            "question":"What is the number of rebounds for team 1?"}
    # This doesn't work
    # result = predictor.predict_json(
    #     table=table,
    #     question="What is the number of rebounds for team 1?"
    # ) 

    # This doesn't work
    # predictor.predict(
    result = predictor.predict_json(input_data)
    print(result['answer'][0]) # First element of 'answer' from result
    # (list (number 19))

if __name__ == "__main__":
    main()
```

**Comment by pdasigi** [comment]
@shubhamagarwal92 The columns are expected to be tab-separated, and rows newline-separated. The code that reads your input table is [here](https://github.com/allenai/allennlp/blob/master/allennlp/semparse/contexts/table_question_knowledge_graph.py#L117). [Here](https://github.com/allenai/allennlp/blob/master/allennlp/semparse/contexts/table_question_knowledge_graph.py#L103-L112) is an example.

I'm currently in the process of replacing this model with the newer model that I mentioned in #2598 (it took longer for me to get to it than expected).  I have an open PR for that here: #2764 that I expect to merge with master in the next couple of days. The newer model will have about 44% accuracy on the official test set.

**Comment by shubhamagarwal92** [comment]
Thanks for the prompt reply. If you upload your SOTA model, kindly keep me updated.

**Comment by schmmd** [comment]
@shubhamagarwal92 I'm going to close this issue since @pdasigi answered your question.
